src/document/mod.rs

```diff
@@ -84,113 +116,225 @@ impl DocumentEncryptOpts {
 }
 
 impl Default for DocumentEncryptOpts {
-    /// default to only sharing with the creator of the document
+    /// Constructs a `DocumentEncryptOpts` with common values.
+    ///
+    /// The document will have a generated ID and no name. Access to the document will be granted only to its creator.
     fn default() -> Self {
         DocumentEncryptOpts::with_explicit_grants(None, None, true, vec![])
     }
 }
 
 #[async_trait]
 pub trait DocumentOps {
-    /// List all of the documents that the current user is able to decrypt.
+    /// Encrypts the provided document bytes.
+    ///
+    /// This will create an entry for the document in the IronCore service. Returns a `DocumentEncryptResult`
+    /// which contains document metadata as well as the `encrypted_data`, which is the only thing that must be passed
+    /// to [document_decrypt()](trait.DocumentOps.html#tymethod.document_decrypt) in order to decrypt the document.
+    ///
+    /// # Arguments
+    /// - `document_data` - Bytes of the document to encrypt
+    /// - `encrypt_opts` - Document encryption parameters. Default values are provided with `DocumentEncryptOpts::default()`.
     ///
-    /// # Returns
-    /// `DocumentListResult` struct with vec of metadata about each document the user can decrypt.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// use ironoxide::document::DocumentEncryptOpts;
+    /// let data = "secret data".as_bytes();
+    /// let encrypted = sdk.document_encrypt(data, &DocumentEncryptOpts::default()).await?;
+    /// # Ok(())
+    /// # }
```

Comment:
Just curious, what do the `#` prefixes mean vs the lines that don't have them?

Reply:
The `#` hides the line from the example. So boilerplate code that will be included in all of them (like the `async fn run`) or things where we're faking the typechecker on (like the `unimplemented!()`) won't have to show up for the users.